### DISCUSSION TITLE
Possible to separate the LSP and yaml completions features from the job monitoring part?

### DISCUSSION THREAD

**Discussion by anentropic** [cschleiden/vscode-github-actions#218] [Ideas]
I saw a strong recommendation online for this extension, based on it's helpers for editing actions YAML files.

But when installing the extension it wants to link to my GitHub profile and requests quite extensive permissions, that I'm not really comfortable giving.

I understand that is because of the other part of the extension, which allows monitoring workflow runs from within VS Code. That is a nice feature but one that I don't have so much interest in.

I was wondering if it might make sense to split these features into two separate extensions?

**Comment by cschleiden** [comment]
Hey, 

Some value of the language server part comes from being able to make calls to github.com (or your server instance). For example, in order to validate references to actions via `uses` and the passed parameters (`with`) we do need to fetch information about the referenced action. Similar for completion and validation of contexts like `secrets`, the extension does need to know what secrets are available. Without that functionality, the value is reduced quite a bit. 

That being said, we have been thinking about releasing the language server part separately in the future. However, I can't make any commitments if and when this might happen.
